### DISCUSSION TITLE
fast dds gen  install failed

### DISCUSSION THREAD

**Discussion by gitguoqun** [eProsima/Fast-DDS-Gen#388] [Q&A] [answered]
Here is a error when i try to install fast dds gen tools .
![屏幕截图 2024-08-01 160001](https://github.com/user-attachments/assets/7d2e3936-c671-482a-8777-f0c179bd6921)
for this error i used "git init" but another error cames out "task: buildIDLParser FAILED"

My ubuntu version is 22.04
My java version is 11
![图片1](https://github.com/user-attachments/assets/372f7249-4d6f-4b4a-920e-2e0b87f133ce)
My IDL-Parser version is 3.0.0,by the way i used ./gradlew assemble to install IDL-Parser. I do not know is this a crrect way?
![图片2](https://github.com/user-attachments/assets/f9e6ba67-9067-4e71-b051-4b47839a59ef)

**Comment by JesusPoderoso** [comment]
Hi @gitguoqun, thanks for using Fast DDS-Gen.
Have you tried the steps of the [installation guide](https://fast-dds.docs.eprosima.com/en/latest/installation/sources/sources_linux.html#fast-dds-gen-installation)?
Go to your clean workspace and try simply this:
```
git clone --recursive https://github.com/eProsima/Fast-DDS-Gen.git fastddsgen
cd fastddsgen
./gradlew assemble
```

**Reply by gitguoqun** [reply]
Hi @JesusPoderoso  Thanks for your replay.  I download the fast dds gen source code in git web and install it. Is that installaion way is not avaliable? Does it must git clone it and then install?

**Reply by JesusPoderoso** [reply, marked as answer]
Fast DDS-Gen contains other repository code as submodules in the "thirdparty" folder.
The key point is the `--recursive` parameter in the `git clone` command.
If you perform the download in git web, the submodules are not initialized, so you would need to setup them manually (`git submodule update --init --recursive`).
